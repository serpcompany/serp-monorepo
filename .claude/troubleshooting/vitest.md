https://vitest.dev/guide/common-errors.html