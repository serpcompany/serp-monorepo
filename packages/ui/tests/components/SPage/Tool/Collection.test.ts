import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import SPageToolCollection from '../../../../components/SPage/Tool/Collection.vue';
import ComponentRender from '../../../componentRender';

mockNuxtImport('useSeoMeta', () => () => {});

describe('SPageToolCollection Snapshot', () => {
  it('renders the default snapshot correctly', async () => {
    const html = await ComponentRender(
      'SPageToolCollection',
      {},
      SPageToolCollection
    );
    expect(html).toMatchSnapshot();
  });
});
