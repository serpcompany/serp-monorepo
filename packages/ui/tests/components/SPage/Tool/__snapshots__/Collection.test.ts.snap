// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageToolCollection Snapshot > renders the default snapshot correctly 1`] = `
<div class="container">
  <div class="
    md:px-8
    md:py-20
    px-4
    py-10
    sm:px-6
    sm:py-16
  ">
    <section class="py-20">
      <h1 class="md:text-center">
        Tools
      </h1>
      <p class="
        md:text-center
        sm:text-md
        text-base
        text-left
      ">
        Free online tools and apps for combining, converting, compressing &amp; convenience.
      </p>
    </section>
    <section class="
      md:container
      md:mx-auto
    ">
      <div class="
        gap-4
        grid
        grid-cols-1
        md:grid-cols-3
        sm:grid-cols-2
      ">
        <!-- Col 1 -->
        <div>
          <h2 class="
            mb-4
            text-xs
            uppercase
          ">
            Combine
          </h2>
          <ul>
            <li>
              <a
                class="
                  flex
                  items-center
                "
                href="/tools/combine-csv-files/"
              > Combine CSVs </a>
            </li>
          </ul>
        </div>
        <!-- Col 2 -->
        <div>
          <h2 class="
            mb-4
            text-xs
            uppercase
          ">
            Convert
          </h2>
          <ul>
            <li>
              <a
                class="
                  flex
                  items-center
                "
                href="/tools/convert-json-to-csv/"
              > JSON to CSV </a>
            </li>
          </ul>
        </div>
        <!-- Col 3 -->
        <div>
          <h2 class="
            mb-4
            text-xs
            uppercase
          ">
            Counters
          </h2>
          <ul>
            <li class="mb-2">
              <a
                class="
                  flex
                  items-center
                "
                href="/tools/count-characters/"
              > Character Counter </a>
            </li>
            <li class="mb-2">
              <a
                class="
                  flex
                  items-center
                "
                href="/tools/paragraph-counter/"
              > Paragraph Counter </a>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>
  <section
    class="
      isolate
      relative
      text-center
    "
    data-orientation="vertical"
  >
    <div class="
      flex
      flex-col
      gap-8
      lg:grid
      lg:px-8
      lg:py-32
      max-w-(--ui-container)
      mx-auto
      px-4
      py-16
      sm:gap-16
      sm:px-6
      sm:py-24
      w-full
    ">
      <div class="">
        <!-- v-if -->
        <!-- v-if -->
        <h2 class="
          font-bold
          lg:text-5xl
          sm:text-4xl
          text-3xl
          text-center
          text-highlighted
          text-pretty
          tracking-tight
        ">
          Subscribe to the newsletter
        </h2>
        <div class="
          mt-6
          sm:text-lg
          text-balance
          text-base
          text-center
          text-muted
        ">
          Join a trillion other readers getting the best info on AI &amp; technology and stay ahead of the curve.
        </div>
        <!-- v-if -->
        <!-- v-if -->
      </div>
      <div class="
        flex
        justify-center
      ">
        <!-- For white background -->
        <button
          class="
            aria-disabled:bg-primary
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            bg-(--ui-background)
            border
            border-neutral-800
            dark:bg-(--ui-background)
            dark:border-white
            dark:hover:bg-neutral-700
            dark:text-white
            disabled:bg-primary
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:outline-2
            focus-visible:outline-offset-2
            focus-visible:outline-primary
            focus-visible:ring-(--ui-primary)
            focus-visible:ring-2
            focus:outline-hidden
            font-semibold
            gap-2
            hover:bg-neutral-100
            inline-flex
            items-center
            my-4
            px-8
            py-3
            rounded-md
            text-lg
            text-neutral-800
            transition-colors
          "
          type="button"
        >
          <!-- v-if -->
          <span class="truncate">
            Subscribe
          </span>
          <!-- v-if -->
        </button>
        <!-- v-if -->
        <!-- teleport start -->
        <!-- teleport end -->
      </div>
    </div>
  </section>
</div>
`;
